Implement Item Disable

Meltie2013 opened the pull request:
<!-- First of all, THANK YOU for your contribution. -->
### Changes proposed

- Add item disabling for Auction House (Player) & Auction House (Bot)
- Add item disabling for Loot Drops. This will allow to remove drops without removing the item from loot table.
- Add item disabling for Vendors. This removes the item from appearing in the vendor list but remains in the database.

### How to Test the Changes:
<!-- Describe in a detailed step-by-step order how to test the changes -->

1. Use the SQL script below to test with the Game Master items. Testing only for Auction House, since it will be the easier to test. Make sure to reload your disable table before testing once the SQL script is added to your database.

Try adding the items listed below to the Auction House after you apply the SQL for Game Master items.

```sql
DELETE FROM disables WHERE sourcetype = 9;
INSERT INTO disables (sourcetype, entry, flags, comment) VALUES
(9,  2586, 3, 'Disable item 2586 (GameMaster\'s Robe) for Auction House Player & Bot'),
(9, 11508, 3, 'Disable item 11508 (GameMaster\'s Slippers) for Auction House Player & Bot'),
(9, 12064, 3, 'Disable item 12064 (GameMaster\'s Hood) for Auction House Player & Bot'),
(9,  7192, 2, 'Disable item 7192 (Plans: Goblin Rocket Boots) for Auction House Bot');
```

### Newly added item disable flags.
```
ITEM_DISABLE_AUCTIONHOUSE     = 0x01
ITEM_DISABLE_AUCTIONHOUSE_BOT = 0x02
ITEM_DISABLE_LOOT_DROP        = 0x04
ITEM_DISABLE_VENDOR           = 0x08
```

### Example Commands:
The idea is to make managing your server easier in the long run then by the old method. 
```
disable add item 926 1 -- This will disable Battle Axe from being sold in the auction house by player
disable add item 926 2 -- This will disable Battle Axe from being sold in the auction house by bot
disable add item 926 3 -- This will disable Battle Axe from being sold in the auction house by player & bot
```
To remove the item from being disable system, use the remove option and all data relating to item 926 will be removed.
```
disable remove item 926
```

### Known issues and TODO list
<!-- Is there anything else left to do after this PR? -->

- [x] Make the needed changes in the database with SQL update for new command


**Notice**: This feature came from MinaseCore my private project I been working on for +10 years. If you plan on to use this feature in your own project, please give credit where its due.

**Disclaimer:** I am releasing this feature under the GPL Version 2.0 license and not AGPL 3.0 license. If you wanting to release this under AGPL 3.0 license, contact me in discord here: https://discord.com/invite/GZ5rsxumxN.

---

- Thank you for collaborating with the project.
- For an efficient working methodology.
- It is obvious to make long pull request, unless it is necessary.
- A short pull request would be easy to test and approve.

Meltie2013 committed:
[Conditions] Implement Item Disable

acidmanifesto reviewed (approved):
Seems alot cleaner this way then thru conf.

Meltie2013 committed:
Finish implementing item disable command

Meltie2013 committed:
Adding items to disable item for GM items and Plans: Goblin Rocket Boots
- GM items are disabled for both player and auction house bot
- Goblin Rocket Boots is disabled for auction house bot

Note: Auction House Bot has been listing GM items as well, so that is fixed.

acidmanifesto reviewed (approved):
Lgtm. We move the auction house disabled out of the conf into a sql table. Makes accidential white space in the conf nonexsistent.

Meltie2013 commented:
@seobryn This PR is ready to be merged.
